### PR TITLE
Added complex_double_get function with test.

### PR DESCRIPTION
**Pull request by ShikharJ**
Relevant: Issue #1088  Feature 3.
Please suggest any changes if necessary.

### CODE REVIEW

**Comment by ShikharJ** [comment]
@isuruf Ping.

File: symengine/cwrapper.h [hunk 1]
```diff
@@ -201,6 +208,9 @@ CWRAPPER_OUTPUT_TYPE complex_imaginary_part(basic s, const basic com);
 CWRAPPER_OUTPUT_TYPE complex_double_real_part(basic s, const basic com);
 //! Assign to s, an imaginary double where com is a complex double
 CWRAPPER_OUTPUT_TYPE complex_double_imaginary_part(basic s, const basic com);
+//! Extract the real and imaginary doubles from the std::complex<double> stored
+//! in basic
+CWRAPPER_OUTPUT_TYPE complex_double_get(dcomplex *d, basic s);
```
Comment (isuruf) [inline symengine/cwrapper.h]: Instead of using pointers, you can return the value. 

    dcomplex complex_double_get(const basic s);

**Comment by ShikharJ** [comment]
@isuruf Something like this?
```
dcomplex complex_double_get(basic s)
{
    SYMENGINE_ASSERT(is_a<ComplexDouble>(*(s->m)));
    dcomplex d;
    d.real = ((rcp_static_cast<const ComplexDouble>(s->m))->as_complex_double()).real();
    d.imag = ((rcp_static_cast<const ComplexDouble>(s->m))->as_complex_double()).imag();
    return d;
}
```

**Comment by isuruf** [comment]
Yes.

    (rcp_static_cast<const ComplexDouble>(s->m))->as_complex_double()

can be replaced with

    static_cast<const ComplexDouble &>(*(s->m)).as_complex_double()

**Comment by ShikharJ** [comment]
My local repository again showed some errors. I'll send another PR with the changes.

**Comment by isuruf** [comment]
@ShikharJ, you don't need to close and send another PR. You can send it to this PR. It;s helpful to have the reviewer comments since they won't be in the new PR

**Comment by ShikharJ** [comment]
@isuruf This is really a mess. It would be better to open a fresh PR in my opinion.

**Pull request closed without merging**